### PR TITLE
Updating email key validation

### PR DESCRIPTION
**Pull request by jland-redhat**
PR in relationship to this issue:
https://github.com/terraform-redhat/terraform-provider-rhcs/issues/341

Just removed the validation since I don't believe it is correct.

Removed the test that was "testing" the validation of the email key. 

And modified some of the other test values so they are closer to expected values.

### CODE REVIEW

**Comment by bardielle** [comment]
I saw that in rosa cli there is no validation on email, however, I think it's a important to keep it in the terraform provider, @ciaranRoche what do you think?

**Comment by jland-redhat** [comment]
But the validation is not correct. You are validating the LDAP **key** not the actual value.

**Comment by jland-redhat** [comment]
Also pushed a fix to the unused import

File: provider/identityprovider/ldap.go [hunk 1]
```diff
@@ -74,11 +73,7 @@ var ldapAttrSchema = map[string]schema.Attribute{
 		Description: "The list of attributes whose values should be used as the email address.",
 		ElementType: types.StringType,
 		Optional:    true,
-		Validators: []validator.List{
-			listvalidator.ValueStringsAre(
-				stringvalidator.RegexMatches(common.EmailRegexp, "Invalid email"),
-			),
-		},
+		Computed:    true,
```
Comment (bardielle) [inline provider/identityprovider/ldap.go:76]: I'm not sure I understood - why did you change it to be computed?

**Comment by ciaranRoche** [comment]
> I saw that in rosa cli there is no validation on email, however, I think it's an important to keep it in the terraform provider, @ciaranRoche what do you think?

Feel like this would be better handled on the service side WDYT? 

I would like to remove any split brain issues here, if you folks want to keep it I would suggest adding the validation to https://github.com/openshift-online/ocm-common/tree/main/pkg/idp/validations so we can add it to CS and not maintain the logic in two places.

File: provider/identityprovider/ldap.go (hunk 1 above)
Comment (jland-redhat) [reply provider/identityprovider/ldap.go:76]: Honestly not 100% sure what computed does here, but I figured it would make sense to be consistent with each of the other LDAP fields which have computed set to true


For example:
```
	"id": schema.ListAttribute{
		Description: "The list of attributes whose values should be used as the user ID. (default ['dn'])",
		ElementType: types.StringType,
		Optional:    true,
		Computed:    true,
	},
	"name": schema.ListAttribute{
		Description: "The list of attributes whose values should be used as the display name. (default ['cn'])",
		ElementType: types.StringType,
		Optional:    true,
		Computed:    true,
	},
```

Comment (sagidayan) [reply provider/identityprovider/ldap.go:76]: now I am wondering if they all should not be computed... @jland-redhat Did you test this manually? With an actual LDAP service?

**Comment by sagidayan** [comment]
Overall, looks good. I want to keep it open to get to the bottom of the LDAP `computed` attrs.

/ok-to-test

**Comment by sagidayan** [comment]
@JohnStrunk can we have your input on this PR and the `computed` attrs?

**Comment by JohnStrunk** [comment]
> @JohnStrunk can we have your input on this PR and the `computed` attrs?

Looking through the idp code, it looks like e-mail should have `computed: true`. Having `computed` set means we can assign the state value based on what we calculate/receive from the OCM API. For example, in Read(), which is triggered from Import(), we set the e-mail based on what is returned. This is important so that we can pick up on changes that are made via the UI (and other non-TF methods).

**Comment by bardielle** [comment]
@jland-redhat please also change the population to be like @JohnStrunk mentioned [here](https://github.com/terraform-redhat/terraform-provider-rhcs/blob/ced64a8c53b28e23ee7ad817970d3a61e31abd02/provider/identityprovider/identity_provider_resource.go#L488)
```
				email := attributes.Email()
				state.LDAP.Attributes.EMail, err = common.StringArrayToList(email)
				if err != nil {
					response.Diagnostics.AddError("failed to convert LDAP attribute EMail to tf list", err.Error())
				}

```

**Comment by bardielle** [comment]
I meant that you can change it not to use the "get" function but set it with the response from the backend by that:
`email := attributes.Email()` instead of `email, ok := attributes.GetEmail()`

@jland-redhat I think it's fine to leave just like that since all the computed attributes were set the same with the `get` function. 

Please fix the subsystem tests 

/lgtm
